addons/controllers/csi/vspherecsiconfig_controller.go

```diff
@@ -142,8 +146,8 @@ func (r *VSphereCSIConfigReconciler) SetupWithManager(_ context.Context, mgr ctr
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *VSphereCSIConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.Log = r.Log.WithValues("VSphereCSIConfig", req.NamespacedName)
-	ctx = logr.NewContext(ctx, r.Log)
+	l := r.Log.WithValues("VSphereCSIConfig", req.NamespacedName)
```

Comment:
This is to workaround a race condition in the test, since the event handler also use r.Log at the same time